saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/SavedSitesSyncDataObserver.kt

```diff
@@ -38,28 +44,40 @@ import timber.log.Timber
 class SavedSitesSyncDataObserver @Inject constructor(
     private val savedSitesRepository: SavedSitesRepository,
     private val syncEngine: SyncEngine,
-    private val deviceSyncState: DeviceSyncState,
+    private val syncStateMonitor: SyncStateMonitor,
     @AppCoroutineScope private val coroutineScope: CoroutineScope,
-    private val dispatcherProvider: DispatcherProvider,
+    private val dispatchers: DispatcherProvider
 ) : MainProcessLifecycleObserver {
 
-    private var initialised: Boolean = false
+    private var dataObserverJob: Job? = null
 
     override fun onCreate(owner: LifecycleOwner) {
         super.onCreate(owner)
-        if (deviceSyncState.isUserSignedInOnDevice()) {
-            Timber.d("Sync-Feature: Listening to changes in Saved Sites")
-            coroutineScope.launch(dispatcherProvider.io()) {
+        syncStateMonitor.syncState()
+            .onEach { state ->
+                when (state) {
+                    OFF -> cancelSavedSitesChanges()
+                    else -> observeSavedSitesChanges()
+                }
+            }
+            .launchIn(coroutineScope)
```

Comment:
Are we never cancelling this subscription to the `syncState()` flow?

Reply:
What do you think won’t work? We listen for changes in the sync state, if it’s disabled we cancel the job that listens for changes, if it’s enabled we listen.